include/SFML/Window/GlResource.hpp

```diff
@@ -48,23 +50,24 @@ class SFML_WINDOW_API GlResource
     GlResource();
 
     ////////////////////////////////////////////////////////////
-    /// \brief Destructor
+    /// \brief Register an OpenGL object to be destroyed when its containing context is destroyed
+    ///
+    /// This is used for internal purposes in order to properly
+    /// clean up OpenGL resources that cannot be shared bwteen
+    /// contexts.
+    ///
+    /// \param object Object to be destroyed when its containing context is destroyed
     ///
     ////////////////////////////////////////////////////////////
-    ~GlResource();
+    static void registerUnsharedGlObject(std::shared_ptr<void> object);
```

Comment:
`registerPrivateGlObject()`?

Reply:
Private is different from not being shared. Some OpenGL objects like FBOs can't be shared among different contexts. They can however still be accessed by anyone who knows which context to activate to use them, so if you want to argue from an access control perspective they are still public.